fixed ClassCastException when retrieving bean

musketyr opened the pull request:
fixes #1851 (again)

graemerocher commented:
How does one reproduce the error?

musketyr commented:
@graemerocher you can use https://github.com/musketyr/mn-intro-problem

graemerocher commented:
The issue is isolating it in an actual example detached from that which is quite complex

musketyr commented:
I'll try to update your test to make it failing

musketyr commented:
@graemerocher there might something change meanwhile since `1.1.4` so I can't get the exact exception but it's wasn't working anyway so I've updated the test and added another fix
